docs/source/reference/kubernetes/kubernetes-deployment.rst

```diff
@@ -0,0 +1,235 @@
+.. _kubernetes-deployment:
+
+Deployment Guides
+-----------------
+Below we include minimal guides to set up a new Kubernetes cluster in different environments, including hosted services on the cloud.
+
+.. grid:: 2
+    :gutter: 3
+
+    .. grid-item-card::  Local Development Cluster
+        :link: kubernetes-setup-kind
+        :link-type: ref
+        :text-align: center
+
+        Run a local Kubernetes cluster on your laptop with ``sky local up``.
+
+    .. grid-item-card::  On-prem Clusters (RKE2, K3s, etc.)
+        :link: kubernetes-setup-onprem
+        :link-type: ref
+        :text-align: center
+
+        For on-prem deployments with kubeadm, RKE2, K3s or other distributions.
+
+    .. grid-item-card:: Google Cloud - GKE
+        :link: kubernetes-setup-gke
+        :link-type: ref
+        :text-align: center
+
+        Google's hosted Kubernetes service.
+
+    .. grid-item-card::  Amazon - EKS
+        :link: kubernetes-setup-eks
+        :link-type: ref
+        :text-align: center
+
+        Amazon's hosted Kubernetes service.
+
+.. _kubernetes-setup-kind:
+
+
+Deploying locally on your laptop
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To try out SkyPilot on Kubernetes on your laptop or run SkyPilot
+tasks locally without requiring any cloud access, we provide the
+:code:`sky local up` CLI to create a 1-node Kubernetes cluster locally.
+
+Under the hood, :code:`sky local up` uses `kind <https://kind.sigs.k8s.io/>`_,
+a tool for creating a Kubernetes cluster on your local machine.
+It runs a Kubernetes cluster inside a container, so no setup is required.
+
+1. Install `Docker <https://docs.docker.com/engine/install/>`_ and `kind <https://kind.sigs.k8s.io/>`_.
+2. Run :code:`sky local up` to launch a Kubernetes cluster and automatically configure your kubeconfig file:
+
+   .. code-block:: console
+
+     $ sky local up
+
+3. Run :code:`sky check` and verify that Kubernetes is enabled in SkyPilot. You can now run SkyPilot tasks on this locally hosted Kubernetes cluster using :code:`sky launch`.
+4. After you are done using the cluster, you can remove it with :code:`sky local down`. This will terminate the KinD container and switch your kubeconfig back to it's original context:
```

Comment:
KinD container is not mentioned in previous parts, making it a bit hard to understand. How about we just call it `kubernetes cluster`?
```suggestion
4. After you are done using the cluster, you can remove it with :code:`sky local down`. This will destroy the local kubernetes cluster and switch your kubeconfig back to it's original context:
```